client.go

```diff
@@ -1233,6 +1435,40 @@ func nodePathToURL(node *Node, path string) url.URL {
 	}
 }
 
+func (c *InternalHTTPClient) Hosts(ctx context.Context) ([]struct {
```

Comment:
we just need to be aware that future code (like that on `cluster-resize`) would use `URI` here.